fix: replication method to proper enum and remove duplicate methods

pnadolny13 opened the pull request:
- The SDK cares about replication method being uppercased when it does a compare in https://github.com/meltano/sdk/blob/5f1a68bcbe68a58c08d7525d7c2b76a69a689a4f/singer_sdk/streams/core.py#L751. I updated these to be uppercase. I assume previously they were skipping this logic, and maybe others ,by accident.
- the `get_url_params` method is defined on the base client class then re-implemented with no changes in a bunch of these streams. I removed the re-implementations where they werent needed.

edgarrmondragon commented on tap_facebook/streams.py:
```diff
@@ -98,7 +98,7 @@ class AdsInsightStream(FacebookStream):
 
     replication_keys = ["date_start"]  # noqa: RUF012
-    replication_method = "incremental"
+    replication_method = "INCREMENTAL"
```
We could probably make this even safer by using `singer_sdk.streams.core.REPLICATION_INCREMENTAL`

pnadolny13 commented:
@edgarrmondragon I updated the base branch to be a branch off of main. I have a series of changes that I want to make that might make the tap buggy in the short term. For example I think the fact that I fixed the replication methods here causes some parameters that arent currently allowed.
